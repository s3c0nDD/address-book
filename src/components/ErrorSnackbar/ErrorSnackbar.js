import React from 'react';

import Alert from '@material-ui/lab/Alert';
import Button from '@material-ui/core/Button';
import Snackbar from '@material-ui/core/Snackbar';

const SnackbarAlert = ({ open }) => {
  const doReload = () => {
    window.location.reload()
  };

  return (
    <Snackbar
      open={open}
    >
      <Alert
        variant="filled"
        severity="error"
        action={
          <Button
            color="inherit"
            variant="outlined"
            size="medium"
            onClick={doReload}
          >
            Reload app
          </Button>
        }
      >
        Error occurred, please click button to reload app
      </Alert>
    </Snackbar>
  )
};

export default SnackbarAlert;
