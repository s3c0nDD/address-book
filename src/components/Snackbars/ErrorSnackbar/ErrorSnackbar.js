import React from 'react';
import PropTypes from 'prop-types';

import Button from '@material-ui/core/Button';

import CommonSnackbar from '../CommonSnackbar';

const ErrorSnackbar = ({ open }) => {
  const doReload = () => {
    window.location.reload();
  };

  return (
    <CommonSnackbar
      action={
        <Button
          color="inherit"
          variant="outlined"
          size="medium"
          onClick={doReload}
        >
          Reload app
        </Button>
      }
      open={open}
      message="Error occurred, please click button to reload app"
      severity="error"
    />
  )
};

export default ErrorSnackbar;

ErrorSnackbar.propTypes = {
  open: PropTypes.bool
};