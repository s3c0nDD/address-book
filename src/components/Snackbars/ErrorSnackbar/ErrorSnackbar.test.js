import React from 'react';
import { createMount, createShallow } from '@material-ui/core/test-utils';

import Button from '@material-ui/core/Button';

import ErrorSnackbar from './ErrorSnackbar';

describe('ErrorSnackbar', () => {
  let mount;
  let shallow;
  let component;
  let wrapper;
  let reloadMock;

  beforeAll(() => {
    shallow = createShallow();
    mount = createMount();
    reloadMock = jest.fn();
    jest.spyOn(window.location, 'reload').mockImplementation(reloadMock);
  });

  beforeEach(() => {
    component = (
      <ErrorSnackbar
        open={true}
      />
    );
    wrapper = shallow(component);
  });

  afterAll(() => {
    window.location.reload.mockClear();
  });

  it('shallowly renders and matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls reload handler on action Button click', () => {
    const wrapper = mount(component);
    const buttonWrapper = wrapper.find(Button);
    buttonWrapper.props().onClick();
    expect(reloadMock).toBeCalledTimes(1);
  });
});
