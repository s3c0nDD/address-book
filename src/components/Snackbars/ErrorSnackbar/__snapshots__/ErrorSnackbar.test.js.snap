// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorSnackbar shallowly renders and matches snapshot 1`] = `
<CommonSnackbar
  action={
    <ForwardRef(WithStyles)
      color="inherit"
      onClick={[Function]}
      size="medium"
      variant="outlined"
    >
      Reload app
    </ForwardRef(WithStyles)>
  }
  message="Error occurred, please click button to reload app"
  open={true}
  severity="error"
/>
`;
